examples/routing_service/routing_service_msql_adapter/cpp/RsLog.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ *
+ * RTI grants Licensee a license to use, modify, compile, and create derivative
+ * works of the Software.  Licensee has the right to distribute object form
+ * only for use with RTI products.  The Software is provided "as is", with no
+ * warranty of any type, including any warranty for fitness for any purpose.
+ * RTI is under no obligation to maintain or support the Software.  RTI shall
+ * not be liable for any incidental or consequential damages arising out of the
+ * use or inability to use the software.
+ */
+ 
+#ifndef RS_LOG_HPP_
+#define RS_LOG_HPP_
+
+/**
+ * @internal
+ */
+
+#define RTI_RS_XSTR(x_)           #x_
+#define RTI_RS_STRINGIFY(x_)      RTI_RS_XSTR(x_)
+
+/*****************************************************************************
+ *
+*****************************************************************************/
+#define RTI_RS_LOG_HEAD          "|| %s || I ||> "
+#define RTI_RS_LOG_HEAD_WARNING  "|| %s || W ||> "
+#define RTI_RS_LOG_HEAD_ERROR    "|| %s || E ||> "
+#define RTI_RS_LOG_HEAD_TRACE    "|| %s || T ||> "
+
+/*****************************************************************************
+  
+*****************************************************************************/
+
+/* Error */
+#define RTI_RS_ERROR(msg_) \
+    fprintf(stdout,RTI_RS_LOG_HEAD_ERROR "%s\n", RTI_RS_LOG_ARGS, (msg_));
```

Comment:
replace this with throw dds::core::Error(msg), which will help you simplify the refactoring of error handling

Reply:
I removed RTI_RS_ERROR from the RsLog file and throw dds::core::Error where necessary